api/app/controllers/spree/api/base_controller.rb

```diff
@@ -2,170 +2,177 @@
+          super + [:price, :variant_id, :sku]
+        else
+          super
         end
+      end
 
-        def authenticate_user
-          unless @current_api_user
-            if requires_authentication? && api_key.blank? && order_token.blank?
-              render "spree/api/errors/must_specify_api_key", :status => 401 and return
-            elsif order_token.blank? && (requires_authentication? || api_key.present?)
-              render "spree/api/errors/invalid_api_key", :status => 401 and return
-            else
-              # An anonymous user
-              @current_api_user = Spree.user_class.new
-            end
-          end
+      private
+
+      def set_content_type
+        content_type = case params[:format]
+        when "json"
+          "application/json; charset=utf-8"
+        when "xml"
+          "text/xml; charset=utf-8"
         end
+        headers["Content-Type"] = content_type
+      end
+
+      def load_user
+        @current_api_user = (try_spree_current_user || Spree.user_class.find_by(spree_api_key: api_key.to_s))
+      end
 
-        def load_user_roles
-          @current_user_roles = if @current_api_user
-            @current_api_user.spree_roles.pluck(:name)
+      def authenticate_user
+        unless @current_api_user
+          if requires_authentication? && api_key.blank? && order_token.blank?
+            render "spree/api/errors/must_specify_api_key", :status => 401 and return
+          elsif order_token.blank? && (requires_authentication? || api_key.present?)
+            render "spree/api/errors/invalid_api_key", :status => 401 and return
           else
-            []
+            # An anonymous user
+            @current_api_user = Spree.user_class.new
           end
         end
+      end
 
-        def unauthorized
-          render "spree/api/errors/unauthorized", status: 401 and return
+      def load_user_roles
+        @current_user_roles = if @current_api_user
+          @current_api_user.spree_roles.pluck(:name)
+        else
+          []
         end
+      end
 
-        def error_during_processing(exception)
-          Rails.logger.error exception.message
-          Rails.logger.error exception.backtrace.join("\n")
+      def unauthorized
+        render "spree/api/errors/unauthorized", status: 401 and return
+      end
 
-          error_notifier.call(exception, self) if error_notifier
+      def error_during_processing(exception)
+        Rails.logger.error exception.message
+        Rails.logger.error exception.backtrace.join("\n")
 
-          render text: { exception: exception.message }.to_json,
-            status: 422 and return
-        end
+        error_notifier.call(exception, self) if error_notifier
 
-        def gateway_error(exception)
-          @order.errors.add(:base, exception.message)
-          invalid_resource!(@order)
-        end
+        render text: { exception: exception.message }.to_json,
+          status: 422 and return
```

Comment:
Align the elements of a hash literal if they span more than one line.<br>Use `&&` instead of `and`.